packages/synapse-react-client/src/components/SynapseChat/SynapseChatInteraction.tsx

```diff
@@ -96,10 +110,93 @@ export function SynapseChatInteraction({
   const textContent = useMemo(() => {
     const parser = new DOMParser()
     const doc = parser.parseFromString(chatResponseText ?? '', 'text/html')
+
+    // Remove tool_name elements
     const elementsToRemove = doc.querySelectorAll('tool_name')
     elementsToRemove.forEach(element => element.remove())
+
+    // Handle actions elements with redirect subelements
+    // custom agents are instructed to include these to indicate navigation.
```

Comment:
To reduce the risk of this functionality while it is being prototyped, could we inject this behavior (e.g. write it in the portal's `synapseChatConfig.ts` and pass as a prop/context value)? You also wouldn't have to handle all app routing behavior in one function